include/zephyr/bluetooth/audio/bap.h

```diff
@@ -1315,26 +1313,20 @@ int bt_bap_unicast_client_discover(struct bt_conn *conn, enum bt_audio_dir dir);
 struct bt_bap_base_bis_data {
 	/* Unique index of the BIS */
 	uint8_t index;
-#if defined(CONFIG_BT_AUDIO_CODEC_CFG_MAX_DATA_COUNT)
-	/** Codec Specific Data count.
-	 *
-	 *  Only valid if the data_count of struct bt_audio_codec_cfg in the subgroup is 0
-	 */
-	size_t data_count;
-	/** Codec Specific Data
-	 *
-	 *  Only valid if the data_count of struct bt_audio_codec_cfg in the subgroup is 0
-	 */
-	struct bt_audio_codec_data data[CONFIG_BT_AUDIO_CODEC_CFG_MAX_DATA_COUNT];
-#endif /* CONFIG_BT_AUDIO_CODEC_CFG_MAX_DATA_COUNT */
+#if defined(CONFIG_BT_AUDIO_CODEC_CFG_MAX_DATA_SIZE)
```

Comment:
```suggestion
#if CONFIG_BT_AUDIO_CODEC_CFG_MAX_DATA_SIZE > 0
```
As the option can be defined, but it's value can be 0.